tests/feature_tests/union.c

```diff
@@ -0,0 +1,54 @@
+int main() {
```

Comment:
Great tests! Couple extra things I can think of to add:
- Try a union containing a struct, and a struct containing a union
- Verify the following produces an error:
```
union A;
struct A;
```
- Verify the following also produces an error:
```
union B;
struct B;
```